File: finance_client/src/app/user/data/types/customer-types/stock-category.ts
export interface StockCategory {
  categoryDescription?: string;
  categoryName?: string;
  deletedBy?: string;
  deletedFlag?: string;
  deletedTime?: Date;
  id?: 0;
  modifiedBy?: string;
  modifiedFlag?: string;
  modifiedTime?: Date;
  postedFlag?: string;
  postedTime?: Date;
  postedBy?: string;
  products?: [
    {
      categoryName?: string;
      category_id?: string;
      deletedBy?: string;
      deletedFlag?: string;
      deletedTime?: Date;
      description?: string;
      id?: 0;
      modifiedBy?: string;
      modifiedFlag?: string;
      modifiedTime?: Date;
      name?: string;
      postedFlag?: string;
      postedTime?: Date;
      price?: 0;
      reason?: string;
      status?: string;
      stock?: 0;
      verifiedBy?: string;
      verifiedFlag?: string;
      verifiedTime?: Date;
    }
  ];
  reason?: string;
  status?: string;
  verifiedBy?: string;
  verifiedFlag?: string;
  verifiedTime?: Date;
}

File: finance_server/src/main/java/co/ke/emtechhouse/eims/StockManagement/categories/StockCategory.java
package co.ke.emtechhouse.eims.StockManagement.categories;

import co.ke.emtechhouse.eims.PurchaseOrderComponent.PoParticularsComponent.PoParticulars;
import co.ke.emtechhouse.eims.StockManagement.products.Products;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
@Table(name = "stockcategories")
public class StockCategory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;
    @Column(nullable = false)
    private String categoryName;
    @Column()
    private String categoryDescription;
    private String status = "pending";
    private String reason = "-";

    private Character postedFlag;
    private Date postedTime;
    private String modifiedBy;
    private Character modifiedFlag;
    private Date modifiedTime;
    private String verifiedBy;
    private Character verifiedFlag;
    private Date verifiedTime;
    private String deletedBy;
    private Character deletedFlag;
    private Date deletedTime;

    @OneToMany(targetEntity = Products.class, cascade = CascadeType.ALL)
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    @JsonProperty(access= JsonProperty.Access.READ_ONLY)
    private List<Products> products;
}
